pyaerocom/scripts/cams2_83/evaluation.py

```diff
@@ -0,0 +1,147 @@
+
+
+class EvalType(str, Enum):
+    LONG = "long"
+    SEASON = "season"
+    WEEK = "week"
+    DAY = "day"
+
+    def __str__(self) -> str:
+        return self.value
+
+    def check_dates(self, start_date: date, end_date: date) -> None:
+        if self == "day" and start_date != end_date:
+            raise ValueError(f"For single day {start_date=} and {end_date=} should be the same")
+
+        if self == "week" and (days := (end_date - start_date) // timedelta(days=1)) < 7:
+            raise ValueError(f"For week needs {days=} >= 7")
+
+    def freqs_config(self) -> dict:
+        if self == "long":
+            return dict(
+                freqs=["daily", "monthly"],
+                ts_type="hourly",
+                main_freq="daily",
+                forecast_evaluation=True,
+            )
+
+        if self == "season":
+            return dict(
+                freqs=["hourly", "daily"],
+                ts_type="hourly",
+                main_freq="hourly",
+                forecast_evaluation=True,
+            )
+
+        if self == "week":
+            return dict(
+                freqs=["hourly", "daily"],
+                ts_type="hourly",
+                main_freq="hourly",
+                forecast_evaluation=False,
+            )
+        if self == "day":
+            return dict(
+                freqs=["hourly"],
+                ts_type="hourly",
+                main_freq="hourly",
+                forecast_evaluation=False,
+            )
+
+        raise NotImplementedError(f"Unsupported {self}")
+
+    def periods(self, start_date: date, end_date: date) -> list[str]:
+        if self == "long":
+            return make_period_ys(start_date, end_date)
+        return make_period(start_date, end_date)
+
+
+def date_range(start_date: date, end_date: date) -> tuple[date, ...]:
+    days = (end_date - start_date) // timedelta(days=1)
+    assert days >= 0
+    return tuple(start_date + timedelta(days=day) for day in range(days + 1))
+
+
+def season(date: date) -> Literal["DJF", "MAM", "JJA", "SON"]:
+    return ("DJF", "MAM", "JJA", "SON")[date.month % 12 // 3]
+
+
+def seasons_in_period(start_date: date, end_date: date) -> list[str]:
+    dates = date_range(start_date, end_date)
+    periods = []
+    prev_date = start_period = dates[0]
+    prev_season = season(prev_date)
+    for current_date in dates[1:]:
+        if season(current_date) == prev_season:
+            prev_date = current_date
+        else:
+            periods.append(f"{start_period:%Y%m%d}-{prev_date:%Y%m%d}")
+            prev_date = current_date
+            prev_season = season(current_date)
+            start_period = current_date
+
+    else:
+        if start_period == dates[-1]:
+            periods.append(f"{start_period:%Y%m%d}")
+        else:
+            periods.append(f"{start_period:%Y%m%d}-{dates[-1]:%Y%m%d}")
+
+    return periods
+
+
+def years_starting_in_november(start_date: date, end_date: date) -> list[str]:
```

Comment:
so... I am not sure I get why this is necessary at all, I have been looking into what the periods are in different kind of evaluations and:
- if the type of evaluation is "long" like we have for the multi-seasonal runs done every 3 months, the periods are just a year range + single years, the function that determines them is `make_period_ys` e.g. with `start_date = datetime(2021,6,1)` and `end_date = datetime(2024,2,29)` the periods will be `['2021-2024', '2021', '2022', '2023', '2024']`, so this case is pretty easy
- for the rest of the evaluations the periods list has basically just a single item which is the date-range in most cases, for example with `start_date = datetime(2023,12,28)` and `end_date = datetime(2024,1,12), `make_period(start_date,end_date) = ['20231228-20240112']`  
 unless it's an across-season range and then the function `years_starting_in_november` comes to play,  e.g. with `start_date = datetime(2024,5,28)` and `end_date = datetime(2024,6,12)` we have `make_period(start_date,end_date) = ['20240528-20240612', '20240528-20241130']`   
  because in the function make_period we have the case 
  ```
    periods = [f"{start_date:%Y%m%d}-{end_date:%Y%m%d}"]
    if periods != season_periods:
        periods += nov_periods  # season_periods
  ``` 
  and here  
  `seasons_in_period(start_date, end_date) = ['20240528-20240531', '20240601-20240612']`  
  `years_starting_in_november(start_date, end_date) = ['20240528-20241130']`

Reply:
![Screenshot from 2024-04-22 11-14-51](https://github.com/metno/pyaerocom/assets/89246953/4e5b0748-34a7-43b2-813b-0dc443cc5f52)  
my guess is that it might have to do with the season selection in the drop down menu

Reply:
anyway, I am not sure where and how it would be best to document all this

Reply:
update: I run a week across seasons and this is how the second period looks like 
![Screenshot from 2024-04-22 14-16-01](https://github.com/metno/pyaerocom/assets/89246953/78e55670-a156-403b-b10f-5d752b13dd7a)
I have no idea what's the point of this